LoopKit/DeviceManager/DeviceLog/PersistentDeviceLog.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  PersistentDeviceLog.swift
+//  LoopKit
+//
+//  Created by Pete Schwamb on 1/13/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import CoreData
+import os.log
+
+
+// Using a framework specific class will search the framework's bundle for model files.
+class PersistentContainer: NSPersistentContainer { }
+
+public class PersistentDeviceLog {
+    
+    private let storageFile: URL
+    
+    private let managedObjectContext: NSManagedObjectContext
+
+    private let persistentContainer: NSPersistentContainer
+    
+    private let maxAge: TimeInterval
+    
+    private var earliestLogDate: Date {
+        return Date(timeIntervalSinceNow: -maxAge)
+    }
+    
+    private let log = OSLog(category: "PersistentDeviceLog")
+    
+    public init(storageFile: URL, maxAge: TimeInterval = TimeInterval(7 * 24 * 60 * 60)) {
```

Comment:
The TimeInterval helpers are internal visibility, so we can't use them in default arg values.  As for duration, I picked a value that covers at least two pod changes. We can easily change this from Loop if the requirement ends up being higher. I did check our Tidepool Loop logging document, and we didn't note any specific duration in it. Right now the main log sharing format is via an issue report. It will start to get unwieldy if we make this too long.